src/components/remote-promo-sheet/check-fns/hasNonZeroAssetBalance.ts

```diff
@@ -0,0 +1,34 @@
+import store from '@/redux/store';
+import { RainbowNetworks } from '@/networks';
+import { ethereumUtils } from '@/utils';
+import { EthereumAddress } from '@/entities';
+
+export const hasNonZeroAssetBalance = async (
+  assetAddress: EthereumAddress
+): Promise<boolean> => {
+  const { selected } = store.getState().wallets;
+  if (!selected) return false;
+
+  const { accountAddress } = store.getState().settings;
+
+  const networks = RainbowNetworks.map(network => network.value);
+
+  // check native asset balances on networks
+  const balancePromises = networks.map(network =>
+    ethereumUtils
+      .getNativeAssetForNetwork(network, accountAddress)
```

Comment:
this function is expecting (network, address) where address is the address of the asset whose balance you want (not the wallet account address)